Add release note for e5.8.1 (#2684)
emqtt/emq-docs-cn: en_US/changes/changes-ee-v5.md

  
  Key improvements include:
  
  - Implemented concurrent RPC calls to retrieve metrics from nodes across the cluster.
  - Introduced data downsampling to reduce the density of data points based on the requested query time span:
    - `10s` intervals for the last `1h`
    - `1m` intervals for the last `1d`
    - `5m` intervals for the last `3d`
    - `10m` intervals for the last `7d`
  - Added dummy data points for periods when EMQX was stopped, ensuring visibility of gaps in the timeline on the dashboard.

#### Enterprise License

- [#13910](https://github.com/emqx/emqx/pull/13910) Performance enhancement for Enterprise edition license check.

### Bug Fixes

#### Core MQTT Functions

- [#13702](https://github.com/emqx/emqx/pull/13702) Clean up the corresponding exclusive subscriptions when a node goes down.

- [#13708](https://github.com/emqx/emqx/pull/13708) Fixed an issue which may cause shared subscription 'sticky' strategy to degrade to 'random'.

- [#13733](https://github.com/emqx/emqx/pull/13733) Made `cacertfile` optional when configuring https listener from `emqx ctl conf load` command.

> - [#13742](https://github.com/emqx/emqx/pull/13742) Fixed when subscribing with `+` as the first level, or `#` as a wildcard, retained messages with topics starting with `$` are incorrectly received.

Reviewer: ```suggestion
- [#13742](https://github.com/emqx/emqx/pull/13742) Fixed an issue when a client would receive retained messages for topic starting with `$` when it subscribed to topic `#` or `+`.
```

- [#13754](https://github.com/emqx/emqx/pull/13754) Fixed an issue when websocket connection would break consistently on its own.

- [#13756](https://github.com/emqx/emqx/pull/13756) Introduced more randomness to broker assigned client IDs.

- [#13790](https://github.com/emqx/emqx/pull/13790) The default heartbeat interval for the MQTT connector has been reduced from 300 seconds to 160 seconds.

  This change helps maintain the underlying TCP connection by preventing timeouts due to the idle limits
  imposed by load balancers or firewalls, which typically range from 3 to 5 minutes depending on the cloud provider.

> - [#13832](https://github.com/emqx/emqx/pull/13832) Fixed that the `Publish` endpoint would have a 500 error when persistent session were enabled.

Reviewer: ```suggestion
- [#13832](https://github.com/emqx/emqx/pull/13832) Fixed 500 error when using `/publish` REST API endpoint with persistent sessions enabled.
```

#### Data Integrations

- [#13851](https://github.com/emqx/emqx/pull/13851) Fixed an exception of `Test connectivity` with the IotDB Thrift driver.

- [#13724](https://github.com/emqx/emqx/pull/13724) Azure Blob Storage and S3 actions in aggregated mode now trigger sending aggregated data sooner after the maximum number of records is reached.

- [#13734](https://github.com/emqx/emqx/pull/13734) Made Azure Blob Storage connector configuration error messages more friendly.

- [#13736](https://github.com/emqx/emqx/pull/13736) Upgraded Kafka producer to support client re-authentication. See [kafka_protocol#122](https://github.com/kafka4beam/kafka_protocol/pull/122).

  Also fixed below minor issuses:

  - `unexpected_info` error log in [PR#13727](https://github.com/emqx/emqx/pull/13727) and [wolff#74](https://github.com/kafka4beam/wolff/pull/74).
  - `einval` crash report of Kafka connection due to a race condition [kafka_protocol#124](https://github.com/kafka4beam/kafka_protocol/pull/124).

- [#13896](https://github.com/emqx/emqx/pull/13896) Upgraded pulsar client from `0.8.3` to `0.8.4` (see [pulsar#61](https://github.com/emqx/pulsar-client-erl/pull/61).

>   Piror this fix, if the producer client experiences a 'socket error' (but not a normal 'socket close'), it may continue sending data to a clsoed socket without any error handling.

Reviewer: ```suggestion
  Prior to this fix, if the producer client experiences a 'socket error' (but not a normal 'socket close'), it may continue sending data to a closed socket without any error handling.
```

>   From EMQX Dashboard, user may observe that the 'totoal' counter keep increasing, but not the other counters: 'success', 'failed' and 'dropped'.

Reviewer: ```suggestion
  From the EMQX Dashboard, user may observe that the 'total' counter keeps increasing, but the other counters 'success', 'failed' and 'dropped' are not.
```
